src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -906,12 +906,12 @@ internal async Task Cleanup(bool async, bool connectionClosing=false)
                 // If the reader is being closed as part of the connection closing, we don't apply
                 // the reader's CommandBehavior.CloseConnection
                 if (_behavior.HasFlag(CommandBehavior.CloseConnection) && !connectionClosing)
-                    _connection.Close();
+                    await _connection.CloseAsync();
```

Comment:
This needs to call Close(bool) and pass async, so that we do sync/async as appropriate

Reply:
done. should the `await _connection.CloseAsync();` on line 914 do the same?

Reply:
Yes, thanks.